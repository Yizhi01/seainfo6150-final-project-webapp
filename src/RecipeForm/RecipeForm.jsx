import React, { useState} from "react";
import styles from "./RecipeForm.module.css";
import Confirmation from "../Confirmation/Confirmation.jsx";

const RecipeForm = () => {
  const [submittedForm, setSubmittedForm] = useState();

  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  }

  return (
    <div>
        {submittedForm ? (
          <Confirmation data={submittedForm}/>
        ) : (
          <form className={styles.recipeForm} onSubmit={onSubmit}>
            <h1 className={styles.formTitle}>Share your recipe with us!</h1>
            <h2 className={styles.sectionTitle}>Recipe Detail</h2>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="title">
                Title<span>*</span>
              </label>
              <input type="text" name="title" id="title" required/>
            </div>
            <div className={styles.categoryGroup}>
              <label className={styles.groupLabel} htmlFor="category">Recipe Category<span>*</span></label>
              <div>
                <select className={styles.dropDown} name="category" id="category">
                  <option value="1">Breadfast</option>
                  <option value="2">Lunch</option>
                  <option value="3">Dinner</option>
                  <option value="4">Desserts</option>
                  <option value="5">Drinks</option>
                  <option value="6">Holiday</option>
                </select>
              </div>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="serves">
                Number of Servings<span>*</span>
              </label>
              <input 
                type="text" 
                name="serves" 
                id="serves" 
                pattern="[0-9]{1,2}" 
                title="Please enter number from 1 to 99"
                required/>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="cookTime">
                Cook Time<span>*</span>
              </label>
              <input type="text" name="cookTime" id="cookTime" required/>
            </div>
            <div className={styles.difficultyGroup}>
              <label className={styles.groupLabel}>Difficulty</label>
              <div className={styles.optionGroup}>
                <div>
                  <input 
                    type="radio" 
                    id="easy" 
                    name="difficulty" 
                    value="easy" 
                    defaultChecked />
                  <label className={styles.redioOption} htmlFor="easy">Easy</label>
                </div>
                <div>
                  <input type="radio" id="medium" name="difficulty" value="medium" />
                  <label className={styles.redioOption} htmlFor="medium">Medium</label>
                </div>
                <div>
                  <input type="radio" id="hard" name="difficulty" value="hard"/>
                  <label className={styles.redioOption} htmlFor="hard">Hard</label>
                </div>
              </div>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="description">
                Description<span>*</span>
              </label>
              <textarea rows="3" name="description" id="description" required></textarea>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="ingredients">
                Ingredients<span>*</span>
              </label>
              <textarea rows="4" name="ingredients" id="ingredients" required></textarea>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="instructions">
                Instructions<span>*</span>
              </label>
              <textarea rows="4" name="instructions" id="instructions" required></textarea>
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel} htmlFor="image">Recipe Image</label>
              <input className={styles.recipeFormGroup} type="file" name="image" id="image" />  
            </div>
            
            <h2 className={styles.sectionTitle}>Recipe Credit</h2>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel}>First Name<span>*</span></label>
              <input
                type="text"
                required
                pattern="[A-Za-z]{1,20}"
                title="Please enter 1 to 20 letters"
              />
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel}>Last Name<span>*</span></label>
              <input
                type="text"
                required
                pattern="[A-Za-z]{1,20}"
                title="Please enter 1 to 20 letters"
              />
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel}>Email<span>*</span></label>
              <input type="email" required />
            </div>
            <div className={styles.recipeFormGroup}>
              <label className={styles.groupLabel}>
                How would you rate your overall satisfaction with our recipes?</label>
              <div>
                <select className={styles.dropDown}>
                  <option value="5">Extremely satisfied</option>
                  <option value="4">Somewhat satisfied</option>
                  <option value="3">Neither satisfied nor dissatisfied</option>
                  <option value="2">Somewhat dissatisfied</option>
                  <option value="1">Extremely dissatisfied</option>
                </select>
              </div>
            </div>
            
            <input className={styles.button} type="submit" value="Submit your recipe" />
          </form>
        )}
    </div>
  )
};

export default RecipeForm;