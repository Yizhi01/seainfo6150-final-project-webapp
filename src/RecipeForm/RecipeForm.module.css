input[type=text], input[type=email], textarea {
  width: 600px;
  padding: 8px;
  border: 1px solid rgba(0, 0, 0, 0.3);
  outline: none;
}

input[type=text]:focus, input[type=email]:focus, textarea:focus  {
  border: 3px solid #bedbbb;
}

.recipeForm {
  font-size: 1em;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  margin: 50px 100px;
  padding: 15px;
  border-radius: 10px;
  border: 7px solid #92817a;
}

.formTitle {
  color: #335d2d;
  font-size: 3em;
}

.sectionTitle {
  margin: 20px 0 0;
  color: #707070;
  font-size: 1.5em;
}

.recipeFormGroup {
  display: flex;
  flex-direction: column;
  margin: 15px 0;
}

.groupLabel  {
  font-size: 1em;
  font-weight: 600;
  padding: 0 0 10px 0;
}

.groupLabel span {
  color: red;
}

.categoryGroup, .difficultyGroup {
  width: 600px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 15px 0;
}

.dropDown {
  font-size: 1em;
}

.optionGroup {
  font-weight: 400;
  flex-basis: 50%;
  display: flex;
  justify-content: space-between;
}

.button {
  font-size: 1em;
  margin: 30px 0;
  padding: 10px 20px;
  background-color: #59886b;
  color: #fff;
  border: 3px solid #59886b;
  border-radius: 5px;
}

.button:hover {
  background-color: #fff;
  color: #59886b;
  border: 3px solid #59886b;
}


@media (max-width: 850px) {
  input[type="text"], input[type="email"], textarea {
    width: 100%;
  }

  .recipeFormGroup, .categoryGroup, .difficultyGroup  {
    width: 100%;
  }
}

