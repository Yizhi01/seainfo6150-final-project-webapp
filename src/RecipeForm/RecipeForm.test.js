import React from "react";
import { render } from "@testing-library/react";
import RecipeForm from "./RecipeForm.jsx";

it("renders the RecipeForm component correctly", () => {
  const { container } = render(
    <RecipeForm />
  );
  expect(container).toMatchSnapshot();
});
