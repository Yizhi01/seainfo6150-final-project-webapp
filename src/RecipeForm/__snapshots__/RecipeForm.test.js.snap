// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeForm component correctly 1`] = `
<div>
  <div>
    <form
      class="recipeForm"
    >
      <h1
        class="formTitle"
      >
        Share your recipe with us!
      </h1>
      <h2
        class="sectionTitle"
      >
        Recipe Detail
      </h2>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="title"
        >
          Title
          <span>
            *
          </span>
        </label>
        <input
          id="title"
          name="title"
          required=""
          type="text"
        />
      </div>
      <div
        class="categoryGroup"
      >
        <label
          class="groupLabel"
          for="category"
        >
          Recipe Category
          <span>
            *
          </span>
        </label>
        <div>
          <select
            class="dropDown"
            id="category"
            name="category"
          >
            <option
              value="1"
            >
              Breadfast
            </option>
            <option
              value="2"
            >
              Lunch
            </option>
            <option
              value="3"
            >
              Dinner
            </option>
            <option
              value="4"
            >
              Desserts
            </option>
            <option
              value="5"
            >
              Drinks
            </option>
            <option
              value="6"
            >
              Holiday
            </option>
          </select>
        </div>
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="serves"
        >
          Number of Servings
          <span>
            *
          </span>
        </label>
        <input
          id="serves"
          name="serves"
          pattern="[0-9]{1,2}"
          placeholder="Please enter number from 1 to 99"
          required=""
          title="Please enter number from 1 to 99"
          type="text"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="cookTime"
        >
          Cook Time
          <span>
            *
          </span>
        </label>
        <input
          id="cookTime"
          name="cookTime"
          placeholder="e.g. 1 hr 20 mins"
          required=""
          type="text"
        />
      </div>
      <div
        class="difficultyGroup"
      >
        <label
          class="groupLabel"
        >
          Difficulty
        </label>
        <div
          class="optionGroup"
        >
          <div>
            <input
              checked=""
              id="easy"
              name="difficulty"
              type="radio"
              value="easy"
            />
            <label
              class="redioOption"
              for="easy"
            >
              Easy
            </label>
          </div>
          <div>
            <input
              id="medium"
              name="difficulty"
              type="radio"
              value="medium"
            />
            <label
              class="redioOption"
              for="medium"
            >
              Medium
            </label>
          </div>
          <div>
            <input
              id="hard"
              name="difficulty"
              type="radio"
              value="hard"
            />
            <label
              class="redioOption"
              for="hard"
            >
              Hard
            </label>
          </div>
        </div>
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="description"
        >
          Description
          <span>
            *
          </span>
        </label>
        <textarea
          id="description"
          name="description"
          required=""
          rows="3"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="ingredients"
        >
          Ingredients
          <span>
            *
          </span>
        </label>
        <textarea
          id="ingredients"
          name="ingredients"
          required=""
          rows="4"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="instructions"
        >
          Instructions
          <span>
            *
          </span>
        </label>
        <textarea
          id="instructions"
          name="instructions"
          required=""
          rows="4"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
          for="image"
        >
          Recipe Image
        </label>
        <input
          class="recipeFormGroup"
          id="image"
          name="image"
          type="file"
        />
      </div>
      <h2
        class="sectionTitle"
      >
        Recipe Credit
      </h2>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
        >
          First Name
          <span>
            *
          </span>
        </label>
        <input
          pattern="[A-Za-z]{1,20}"
          required=""
          title="Please enter 1 to 20 letters"
          type="text"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
        >
          Last Name
          <span>
            *
          </span>
        </label>
        <input
          pattern="[A-Za-z]{1,20}"
          required=""
          title="Please enter 1 to 20 letters"
          type="text"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
        >
          Email
          <span>
            *
          </span>
        </label>
        <input
          placeholder="example@gmail.com"
          required=""
          type="email"
        />
      </div>
      <div
        class="recipeFormGroup"
      >
        <label
          class="groupLabel"
        >
          How would you rate your overall satisfaction with our recipes?
        </label>
        <div>
          <select
            class="dropDown"
          >
            <option
              value="5"
            >
              Extremely satisfied
            </option>
            <option
              value="4"
            >
              Somewhat satisfied
            </option>
            <option
              value="3"
            >
              Neither satisfied nor dissatisfied
            </option>
            <option
              value="2"
            >
              Somewhat dissatisfied
            </option>
            <option
              value="1"
            >
              Extremely dissatisfied
            </option>
          </select>
        </div>
      </div>
      <input
        class="button"
        type="submit"
        value="Submit your recipe"
      />
    </form>
  </div>
</div>
`;
